src/main/java/cz/cvut/kbss/termit/security/JwtAuthorizationFilter.java

```diff
@@ -82,13 +87,16 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         }
         final String authToken = authHeader.substring(SecurityConstants.JWT_TOKEN_PREFIX.length());
         try {
-            final TermItUserDetails userDetails = jwtUtils.extractUserInfo(authToken);
-            final TermItUserDetails existingDetails = userDetailsService.loadUserByUsername(userDetails.getUsername());
-            SecurityUtils.verifyAccountStatus(existingDetails.getUser());
+            Jwt jwt = jwtDecoder.decode(authToken);
+            final String username = jwt.getSubject();
+            if (username == null || username.isBlank()) {
+                throw new JwtException("Invalid JWT token contents");
+            }
+            final TermItUserDetails existingDetails = userDetailsService.loadUserByUsername(username);
```

Comment:
It seems a bit weird (and wasteful) that user details are loaded by the `TermItJwtDecoder.decode`, thrown away and the loaded again here.